src/main/java/software/amazon/nio/spi/s3/S3Path.java

```diff
@@ -612,10 +615,32 @@ private boolean isEmpty(){
      */
     @Override
     public URI toUri() {
```

Comment:
Can you add a note in the Java doc how this URI might not be equal to the "S3 URI" that will be seen in the S3 console due to the need for the URI to be encoded to avoid illegal characters?